lib/metadataTypes/Automation.js

```diff
@@ -709,6 +710,172 @@ class Automation extends MetadataType {
+        if (!File.existsSync(directory)) {
+            File.mkdirpSync(directory);
+        }
+        const tabled = [];
+        tabled.push(
+            json.steps.map((step, index) => ({
+                title: `Step ${index + 1}`,
+                description: step.name || '-',
+            }))
+        );
+        let maxActivities = 0;
+        for (const step of json.steps) {
+            if (step.activities.length > maxActivities) {
+                maxActivities = step.activities.length;
+            }
+        }
+        for (let activityIndex = 0; activityIndex < maxActivities; activityIndex++) {
+            tabled.push(
+                json.steps.map((step, stepIndex) =>
+                    step.activities[activityIndex]
+                        ? {
+                              i: stepIndex + 1 + '.' + (activityIndex + 1),
+                              name: step.activities[activityIndex].name,
+                              type: step.activities[activityIndex].r__type,
+                          }
+                        : null
+                )
+            );
+        }
+        console.log(json.key, tabled);
+        let output;
+        if (mode === 'md') {
+            output = this._generateDocMd(json, tabled);
+            try {
+                // write to disk
+                await File.writeToFile(directory, filename + '.automation-doc', mode, output);
+            } catch (ex) {
+                Util.logger.error(`Automation.writeDeToX(${mode}):: error | ` + ex.message);
+            }
+        }
+    }
+    /**
+     * Parses metadata into a readable Markdown/HTML format then saves it
+     *
+     * @param {TYPE.BuObject} buObject properties for auth
+     * @param {TYPE.AutomationMap} [metadata] a list of dataExtension definitions
+     * @param {boolean} [isDeploy] used to skip non-supported message during deploy
+     * @returns {Promise.<void>} -
+     */
+    static async document(buObject, metadata, isDeploy) {
+        if (['md', 'both'].includes(this.properties.options.documentType)) {
+            if (!metadata) {
+                metadata = this.readBUMetadataForType(
+                    File.normalizePath([
+                        this.properties.directories.retrieve,
+                        buObject.credential,
+                        buObject.businessUnit,
+                    ]),
+                    true
+                ).automation;
+            }
+            const docPath = File.normalizePath([
+                this.properties.directories.docs,
+                this.definition.type,
+                buObject.credential,
+                buObject.businessUnit,
+            ]);
+            if (!metadata || !Object.keys(metadata).length) {
+                // as part of retrieve & manual execution we could face an empty folder
+                return;
+            }
+            if (!isDeploy) {
+                File.removeSync(docPath);
```

Comment:
prefer async methods :)

Reply:
removed in favor of saving it into the retrieve folder